### PR TITLE
Added PopupMenuDivider, Divider

### PR DESCRIPTION
**Pull request by HansMuller**
Added a Divider class and replaced uses of DrawerDivider with it. 

Popup menus are now lists of PopupMenuEntry items. PopupMenuItem and PopupMenuDivider are PopupMenuEntry subclasses.

PopupMenuDivider builds a Divider.

### CODE REVIEW

File: packages/flutter/lib/src/material/divider.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+class Divider extends StatelessComponent {
+  const Divider({ Key key, this.height: 16.0, this.indent: 0.0, this.color }) : super(key: key);
```
Comment (abarth) [inline packages/flutter/lib/src/material/divider.dart]: IMHO, it's better to have the height assert in the constructor and have the widget be non-const.  Having the constructor be const limits what we can do with this class in the future.

**Comment by abarth** [comment]
LGTM

File: examples/material_gallery/lib/demo/menu_demo.dart [hunk 2]
```diff
@@ -111,7 +111,7 @@ class MenuDemoState extends State<MenuDemo> {
             primary: new Text('An item with a sectioned menu'),
             right: new PopupMenuButton<String>(
               onSelected: showMenuSelection,
-              items: <PopupMenuItem>[
+              items: <PopupMenuEntry<String>>[
                 new PopupMenuItem(
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/menu_demo.dart]: `PopupMenuItem<String>`?

File: packages/flutter/lib/src/material/divider.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  final double height;
+  final double indent;
+  final Color color;
+
+  Widget build(BuildContext context) {
+    assert (height >= 1.0);
```
Comment (Hixie) [inline packages/flutter/lib/src/material/divider.dart]: no space between "assert" and the expression

File: packages/flutter/lib/src/material/popup_menu.dart [hunk 4]
```diff
@@ -26,20 +27,36 @@ const double _kMenuVerticalPadding = 8.0;
...
-class PopupMenuItem<T> extends StatelessComponent {
+abstract class PopupMenuEntry<T> extends StatelessComponent {
+  PopupMenuEntry({ Key key }) : super(key: key);
+
+  double get height;
+  T get value => null;
+  bool get enabled => true;
+}
+
+class PopupMenuDivider<T> extends PopupMenuEntry<T> {
```
Comment (Hixie) [inline packages/flutter/lib/src/material/popup_menu.dart]: I think `PopupMenuDivider` should probably explicitly extend `PopupMenuEntry<dynamic>`. Then, if I'm not mistaken, you'll be able to use it in a `PopupMenuEntry<T>` list (for any value of T).

**Comment by Hixie** [comment]
lgtm

File: examples/material_gallery/lib/demo/menu_demo.dart (hunk 2 above)
Comment (HansMuller) [inline examples/material_gallery/lib/demo/menu_demo.dart]: This list of items contains PopupMenuItems and a PopupMenuDivider

File: packages/flutter/lib/src/material/popup_menu.dart (hunk 4 above)
Comment (HansMuller) [inline packages/flutter/lib/src/material/popup_menu.dart]: Thanks for this, that's much better. I really didn't want to give PopupMenuDivider a type parameter.

File: examples/material_gallery/lib/demo/menu_demo.dart (hunk 2 above)
Comment (Hixie) [inline examples/material_gallery/lib/demo/menu_demo.dart]: I was talking about the next line, not the one you edited. :-)

Comment (HansMuller) [inline examples/material_gallery/lib/demo/menu_demo.dart]: Sorry yes you're right.  All of the menu items in this example should be PopupMenuItem<String>.  Will fix that.

**Pull request merged**